src/test/java/com/teragrep/nlf_01/types/ADFPipelineRunTypeTest.java

```diff
@@ -0,0 +1,276 @@
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.nlf_01.types;
+
+import com.teragrep.akv_01.event.ParsedEvent;
+import com.teragrep.akv_01.event.ParsedEventFactory;
+import com.teragrep.akv_01.event.UnparsedEventImpl;
+import com.teragrep.akv_01.event.metadata.offset.EventOffset;
+import com.teragrep.akv_01.event.metadata.offset.EventOffsetImpl;
+import com.teragrep.akv_01.event.metadata.offset.EventOffsetStub;
+import com.teragrep.akv_01.event.metadata.partitionContext.EventPartitionContext;
+import com.teragrep.akv_01.event.metadata.partitionContext.EventPartitionContextImpl;
+import com.teragrep.akv_01.event.metadata.partitionContext.EventPartitionContextStub;
+import com.teragrep.akv_01.event.metadata.properties.EventProperties;
+import com.teragrep.akv_01.event.metadata.properties.EventPropertiesImpl;
+import com.teragrep.akv_01.event.metadata.properties.EventPropertiesStub;
+import com.teragrep.akv_01.event.metadata.systemProperties.EventSystemProperties;
+import com.teragrep.akv_01.event.metadata.systemProperties.EventSystemPropertiesImpl;
+import com.teragrep.akv_01.event.metadata.systemProperties.EventSystemPropertiesStub;
+import com.teragrep.akv_01.event.metadata.time.EnqueuedTime;
+import com.teragrep.akv_01.event.metadata.time.EnqueuedTimeImpl;
+import com.teragrep.akv_01.event.metadata.time.EnqueuedTimeStub;
+import com.teragrep.akv_01.plugin.PluginException;
+import com.teragrep.rlo_14.Facility;
+import com.teragrep.rlo_14.SDElement;
+import com.teragrep.rlo_14.SDParam;
+import com.teragrep.rlo_14.Severity;
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonReader;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+class ADFPipelineRunTypeTest {
+
+    private ParsedEvent testEvent(
+            final String path,
+            final EventPartitionContext partitionCtx,
+            final EventProperties props,
+            final EventSystemProperties sysProps,
+            final EnqueuedTime enqueuedTime,
+            final EventOffset offset
+    ) {
+        final InputStream is = Assertions.assertDoesNotThrow(() -> Files.newInputStream(Paths.get(path)));
+        final JsonReader reader = Json.createReader(is);
+
+        final JsonObject json = Assertions.assertDoesNotThrow(reader::readObject);
+
+        Assertions.assertDoesNotThrow(is::close);
+        Assertions.assertDoesNotThrow(reader::close);
+
+        return new ParsedEventFactory(
+                new UnparsedEventImpl(json.toString(), partitionCtx, props, sysProps, enqueuedTime, offset)
+        ).parsedEvent();
+    }
+
+    @Test
+    void testIdealCase() {
```

Comment:
there exists https://github.com/teragrep/aer_02/blob/main/src/test/java/com/teragrep/aer_02/FakeParsedEventSource.java in aer_02, should similar be created here so that the fake event generation code is not copied over and over. @eemhu comments please and issue if relevant.